pkg/appendable/index_file.go

```diff
@@ -154,48 +154,48 @@ func (i *IndexFile) IndexFieldNames() ([]string, error) {
 	return fieldNames, nil
 }
 
-func (i *IndexFile) FindOrCreateIndex(name string, fieldType FieldType) (*metapage.LinkedMetaSlot, error) {
+func (i *IndexFile) FindOrCreateIndex(name string, fieldType FieldType) (*metapage.LinkedMetaSlot, uint16, error) {
```

Comment:
I'd return the entire index meta instead.

```suggestion
func (i *IndexFile) FindOrCreateIndex(name string, fieldType FieldType) (*metapage.LinkedMetaSlot, *IndexMeta, error) {
```

Reply:
Yeah that's much more elegant. Thanks